extensions/resteasy-classic/rest-client-config/runtime/src/main/java/io/quarkus/restclient/config/RestClientsBuildTimeConfig.java

```diff
@@ -69,6 +79,33 @@ interface RestClientBuildConfig {
          * </ul>
          */
         Optional<String> localProxyProvider();
+
+        /**
+         * If true, the extension will automatically remove the trailing slash in the paths if any.
+         * This property is not applicable to the RESTEasy Client.
+         */
+        @WithName("removes-trailing-slash")
+        @WithDefault("true")
+        boolean removesTrailingSlash();
+    }
+
+    /**
+     * Get the build config for a concrete REST Client implementation.
+     */
+    default Optional<RestClientBuildConfig> getConfigFor(ClassInfo restClientInterface, Optional<String> configKey) {
```

Comment:
Ah sorry, I missed this was build time config.

Reply:
Yes, we shouldn't be doing this.

Please use https://github.com/quarkusio/quarkus/blob/5c62c8299592a7378d7658ce0448a8598fc069ac/extensions/resteasy-classic/rest-client-config/runtime/src/main/java/io/quarkus/restclient/config/RestClientsBuildTimeConfig.java#L74-L124

Reply:
I tried to use this instead of the provided instance of RestClientsBuildTimeConfig in the build step and as far as I see, the only difference is that the returned map from this method is extended with the clients using the qualified class name, but with the default values. For example, if I have the following REST Interface:

```java
@RegisterRestClient(configKey = "test")
    @Path("/slash/with/")
    public interface HelloClientUsingConfigKey {
        @GET
        @Produces(MediaType.TEXT_PLAIN)
        String echo();
    }
```

And configure it using the following property:

```
quarkus.rest-client.test.removes-trailing-slash=false
```

Using the provided instance of RestClientsBuildTimeConfig.clients(), I get only one entry with key "test" with the correct value of "removes-trailing-slash". And when using the map from the method you suggested, I get two entries: one for the key "test" with the correct value again, and another entry with the key "HelloClientUsingConfigKey", but with the default values. See the following snapshot that illustrates what I'm saying:

![Captura de pantalla de 2024-11-05 14-51-25](https://github.com/user-attachments/assets/169ee84b-1c2d-40bf-ae3a-2eca26e759bf)

Therefore, if the map from this method would return the map with all the REST Interfaces but correct values, I see the value because we could lookup the configuration by class name always. But if not, I don't see how this would help. Can you clarify?

Reply:
The problem with not using `io.quarkus.restclient.config.RestClientsBuildTimeConfig#get` is that it will miss fallbacks / relocations. The previous approach caused several issues detailed in https://github.com/quarkusio/quarkus/issues/41313.

In Build Time, we are not using `@WithKeys` (but we could probably do it), so the map keeps a reference for any entry key found in the configuration, but it will always contain an entry key with the FQN, so that is the key that must be used to access the REST Client configuration.

I've just pushed a small commit to use it, plus a small fix in DevServices that I've found.